src/smc-webapp/buttonbar.coffee

```diff
@@ -51,11 +51,280 @@ exports.commands =
+                insert: 
+                        """
+                        git commit -a -m "commit message"
+                        """ 
+        setup_ssh_for_github :
+                insert: 
+                        """
+                        SSHFILE=~/.ssh/id_rsa
+                        ssh-keygen -t rsa -b 4096 -N "" -C "your_email@example.com" -f $SSHFILE
+                        eval $(ssh-agent -s)
+                        ssh-add ~/.ssh/id_rsa
+                        echo "Below this line is your public SSH key"
+                        cat ~/.ssh/id_rsa.pub
+                        # Copy your public key below and follow the instructions at https://help.github.com/articles/adding-a-new-ssh-key-to-your-github-account/#platform-linux
+                        """ 
+        push_send_changes_to_the_master_branch_of_your_remote_repository :
+                insert:
+                        """
+                        git push origin master
+                        """
+
+        status :
+                insert:
+                        """
+                        git status
+                        """
+
+        connect_to_a_remote_repository :
+                insert:
+                        """
+                        git remote add origin <server>
+                        """
+
+        list_all_currently_configured_remote_repositories :
+                insert:
+                        """
+                        git remote -v
+                        """
+
+        branches_create_a_new_branch_and_switch_to_it :
+                insert:
+                        """
+                        git checkout -b <branchname>
+                        """
+
+        switch_from_one_branch_to_another :
+                insert:
+                        """
+                        git checkout <branchname>
+                        """
+
+        list_all_the_branches_in_your_repo_and_also_tell_you_what_branch_youre_currently_in :
+                insert:
+                        """
+                        git branch
+                        """
+
+        delete_the_feature_branch :
+                insert:
+                        """
+                        git branch -d <branchname>
+                        """
+
+        push_the_branch_to_your_remote_repository_so_others_can_use_it :
+                insert:
+                        """
+                        git push origin <branchname>
+                        """
 
+        push_all_branches_to_your_remote_repository :
+                insert:
+                        """
+                        git push --all origin
+                        """
+
+        delete_a_branch_on_your_remote_repository :
+                insert:
+                        """
+                        git push origin :<branchname>
+                        """
+
+        update_from_the_remote_repository_fetch_and_merge_changes_on_the_remote_server_to_your_working_directory :
```

Comment:
wow, why are the keys so long? leave it if you like it, but a shorter one is better